lib/utils/getImageSizes/index.ts

```diff
@@ -0,0 +1,21 @@
+// Returns a string to be used in the sizes property of the Image component
+// This function assumes that the image layout property equals fill
+// and that the image fills the viewport only if it is width is lower than 768px
+export const getImageSizes = (desktop_width: number) => {
+  const mobile_widths = [256, 384, 640, 768];
+  const mobile_sizes = mobile_widths
+    .map((w) => {
+      return `(max-width: ${widthWithResolution(w)}px) ${w}px`;
+    })
+    .join(", ");
+  return `
+        ${mobile_sizes},
+        ${widthWithResolution(desktop_width)}px
+    `;
+};
+
+const widthWithResolution = (width: number) => {
+  return typeof window !== "undefined"
+    ? Math.round(width * window.devicePixelRatio)
```

Comment:
Yeah, I thought so too. But in practice when I tested with the emulator in chrome I did not get the good values.
To make sure I started a server that I accessed with my phone which is 1080 wide and watch the queries coming to Apache:
With the device ratio trick:
`37.167.48.79 - - [06/Oct/2022:08:53:53 +0200] "GET /_next/image?url=%2F_next%2Fstatic%2Fmedia%2Fgreen-mountainside.14205431.jpg&w=750&q=75 HTTP/1.1" 200 13498`

Without the device ratio trick:
`37.167.48.79 - - [06/Oct/2022:09:10:50 +0200] "GET /_next/image?url=%2F_next%2Fstatic%2Fmedia%2Ftrees-and-water.cfa00b44.png&w=1920&q=75 HTTP/1.1" 200 15268`

I think 750 is the good value since we want to cover 1080 / 2 = 540 pixels. But I am not too good with UI so maybe I am missing something.